### ISSUE TITLE
[VCF] Question on usage

### ISSUE THREAD

**Issue by ironb25** [sbslee/fuc#63]
Follow up question after resolving this(https://github.com/sbslee/fuc/issues/62) on the usage.

1) I want to find common variants, and unique to each of the vcf. (2 vcf files from mutect both)
2) Secondly, I want to concatenate vcf calls from strelka and mutect, and remove duplicates.

Could you suggest which methods to use for these?

thanks,
Rohan

**Comment by sbslee** [comment]
Because I'm currently out of town and don't have my computer with me, please excuse me for not being able to provide the fullest answers. But I will try my best :) I should be back by Sunday (in Korean time), so if below doesn't get you where you need, please let me know and I will try to provide additional support as soon as I get back.

For the first question, there are a number of ways to achieve the goal, but I suggest to use the [`pyvcf.VcfFrame.to_variants()`](https://sbslee-fuc.readthedocs.io/en/latest/api.html#fuc.api.pyvcf.VcfFrame.to_variants) method:

```
from fuc import pyvcf

vf1 = pyvcf.VcfFrame.from_file('in1.vcf')
vf2 = pyvcf.VcfFrame.from_file('in2.vcf')

variants1 = vf1.to_variants()
variants2 = vf2.to_variants()

common_variants = list(set(variants1).intersection(variants2))
unique_variants1 = [x for x in variants1 if x not in variants2]
unique_variants2 = [x for x in variants2 if x not in variants1]
```

For the second question, again, there are a number of ways but I recommend using the [`pyvcf.merge()`](https://sbslee-fuc.readthedocs.io/en/latest/api.html#fuc.api.pyvcf.merge) method with the `collapse=True` option:

```
merged_vf = pyvcf.merge([vf1, vf2], how='outer')
merged_vf.to_file('out.vcf')
```

This will merge two `VcfFrame` objects while removing duplicate records.

**Comment by ironb25** [comment]
The answer to the first question worked , thank you! 

Regarding second question on merging variants from 2 callers,  when doing outer join, it does not work. 
 File "/home/ubuntu/miniconda3/envs/fuc/lib/python3.9/site-packages/fuc/api/pyvcf.py", line 5277, in <listcomp>
    key=lambda col: [CONTIGS.index(x) if isinstance(x, str)
ValueError: 'MT' is not in list

**Comment by sbslee** [comment]
@ironb25,

Thanks for the patience, I'm back! Glad to hear the first question is resolved.

For the second question, the problem occurred because currently the `pyvcf.VcfFrame.sort` method expects you to use either `M` or `chrM` for indicating the mitochondria contig (you are using `MT`). Clearly, this needs to be fixed because people need to be able to use `pyvcf.VcfFrame.sort` regardless of what contigs their input VCF file has. I will update this for the next release `0.34.0` -- it should be an easy fix. In the meantime, can you send me the two VCF files you are trying to merge? This way, I can make sure there are no other surprises for you. But it's totally fine if you can't send them to me.

**Comment by ironb25** [comment]
I shared the file in the email.

**Comment by sbslee** [comment]
@ironb25,

I updated the `pyvcf.VcfFrame.sort` method to handle custom contigs such as `MT`. This resolves your second question. I sent you the output VCF after merging.

FYI, I noticed that the output VCF has no genotype data (e.g. `0/1`) for the samples `NORMAL` and `TUMOR`. When I looked into this more closely, I found that the `strelka2.vcf` file does not have the `GT` field at all. Was this intended?
